src/app/app-cache-handler.ts

```diff
@@ -30,3 +55,29 @@ export const createAppCacheFile = (): void => {
     logger.info(`app-cache-handler: this is a clean exit, creating app cache file`);
     fs.writeFileSync(cacheCheckFilePath, '');
 };
+
+/**
+ * Cleans the app cache on new install
+ */
+export const cleanAppCacheOnInstall = (): void => {
+    logger.info(`app-cache-handler: cleaning app cache and cookies on new install`);
+    cleanOldCache();
+};
+
+/**
+ * Cleans app cache and restarts the app on crash or unresponsive events
+ * @param window Browser window to listen to for crash events
+ */
+export const cleanAppCacheOnCrash = (window: BrowserWindow): void => {
+    logger.info(`app-cache-handler: listening to crash events & cleaning app cache`);
+    const events = ['unresponsive', 'crashed', 'plugin-crashed'];
+
+    events.forEach((windowEvent: any) => {
+        window.webContents.on(windowEvent, () => {
```

Comment:
For `unresponsive` event don't we need user ack?

Reply:
Good point @KiranNiranjan, I'll bring this up tomorrow with Saif.